packages/grafana-ui/src/components/DateTimePickers/TimeRangePicker.tsx

```diff
@@ -66,20 +66,6 @@ export class UnthemedTimeRangePicker extends PureComponent<TimeRangePickerProps,
     this.setState({ isOpen: !isOpen });
   };
 
-  componentDidMount() {
-    window.addEventListener('keyup', this.onKeyUp);
-  }
-
-  componentWillUnmount() {
-    window.removeEventListener('keyup', this.onKeyUp);
-  }
-
-  onKeyUp = (event: KeyboardEvent) => {
-    if (event.code === 'Escape') {
-      this.onClose();
-    }
-  };
-
```

Comment:
also i think removing this completely reintroduces the problem @oscarkilhed was fixing. you can't close the `TimeRangePicker` with the escape key again. think it either needs leaving in, or if it clashes with your code in some way it might need to use `FocusScope`/`useOverlay` in the same way you've done for the `TimePickerCalendar`

Reply:
oh...I get it now 😝
His fixes affects the timePicker as a whole while mine only affects timePickerCalendar!
@oscarkilhed my bad, I totally misunderstood your point

Reply:
I'm currently looking at using useOverlay for this feature instead. As I understand it should then keep track of multiple levels of overlays and only close the last one created when you hit esc. So far I'm stuck on the keyUp event not being propagated to the overlay element.